src/Components/MomentReacter.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState, useRef } from 'react';
+import moment from 'moment';
+import _ from 'lodash';
+
+const FILE_LOG_TAG = 'MomentReacter';
+const MAX_TIMEOUT_TIME = 2147483647;
+interface NextMomentResult {
+    theMoment: moment.Moment;
+    theDiff: number;
+}
+
+const nextMoment = (referenceMoment: moment.Moment, moments: Array<moment.Moment>): NextMomentResult | null => {
+    let result: NextMomentResult | null = null;
+    moments.forEach(theMoment => {
+        // the amount of time until the reference moment, if reference moment is in the future this is positive
+        const theDiff = theMoment.diff(referenceMoment);
+        if (theDiff >= 0 && (_.isNil(result) || theDiff < result.theDiff)) {
+            result = {
+                theMoment,
+                theDiff
+            };
+        }
+    });
+    return result;
+};
+
+interface MomentReacterProps {
+    absolute?: boolean;
+    intervalInMillis?: number;
+    children: (moment: moment.Moment) => JSX.Element;
+    stopMoment?: moment.Moment;
+    significantMoments?: Array<moment.Moment>;
+    offsetInMillis?: number;
+    stop?: boolean;
+    logTag?: string
+}
+
+export const MomentReacter: React.FC<MomentReacterProps> = ({
+    absolute = false,
+    intervalInMillis,
+    children,
+    stopMoment,
+    significantMoments,
+    offsetInMillis,
+    stop,
+    logTag = 'untagged'
+}) => {
+    const TAG = `${FILE_LOG_TAG}---${logTag}`;
+    const [reactiveMoment, setReactiveMoment] = useState<moment.Moment>(moment());
+    const currentTimeoutHandle = useRef<NodeJS.Timeout | null>(null);
+    
+    if(stop === true && !_.isNil(currentTimeoutHandle.current)) {
+        clearTimeout(currentTimeoutHandle.current);
+    }
+    useEffect(() => {
+        if(stop === true) {
+            return;
+        }
+        const currentMoment = moment();
+        let timeoutTime = intervalInMillis;
+
+        if (absolute) {
+            if(_.isNil(intervalInMillis)) {
+                console.error(`${TAG} Cannot use absolute if intervalInMillis is not provided, will use significantMoments if they are provided only`);
+            } else {
+                // We need to mod by the intervalInMillis because this number can be greater
+                // For example let's say we want to run this every minute on the half minute
+                // intervalInMillis = 60000, offsetInMillis = 30000
+                // There is 45 seconds until the next minute, ad the 30 and your at a minute and 15 seconds
```

Comment:
`add` and `you're`

Reply:
lol I'll fix the comments in another PR, otherwise good?